egui_extras/src/table.rs

```diff
@@ -352,24 +352,145 @@ pub struct TableBody<'a> {
 }
 
 impl<'a> TableBody<'a> {
+    fn delta(&self) -> f32 {
+        self.start_y - self.layout.current_y()
+    }
+
+    /// Return a vector containing all column widths for this table body.
+    ///
+    /// This is primarily meant for use with [`TableBody::heterogeneous_rows`] in cases where row
+    /// heights are expected to according to the width of one or more cells -- for example, if text
+    /// is wrapped rather than clippped within the cell.
+    pub fn widths(&self) -> &Vec<f32> {
```

Comment:
This can return a slice.

```suggestion
    pub fn widths(&self) -> &[f32] {
```

Reply:
I'm still fairly new to rust so I don't fully understand the signficance -- is there a reason to prefer one over the other?

Reply:
There a few reasons to return a slice. One is that it provides an abstraction that hides implementation details. For instance the list of widths could internally be stored in any collection which can Deref to slice and the type could be changed freely without changing the API.

Also if you look through the docs for `Vec` you'll see that it only implements very few methods that take `&self`, and the useful ones are also implemented on slice.

Lastly, it's just more idiomatic. In the same way you would pass a slice as an argument instead of a borrowed vector. A slice works with multiple types of collections.

Reply:
That's super helpful to know, thanks for the explanation! I did make the suggested change fyi.

Reply:
Glad to help!